diff --git a/launcher/Launcher.cpp b/launcher/Launcher.cpp
index aba0668a..d0de14bc 100644
--- a/launcher/Launcher.cpp
+++ b/launcher/Launcher.cpp
@@ -521,6 +521,7 @@ Launcher::Launcher(int &argc, char **argv) : QApplication(argc, argv)
         FS::updateTimestamp(m_rootPath);
 #endif
 
+#define MULTIMC_JARS_LOCATION @multimc_jars_location@
 #ifdef MULTIMC_JARS_LOCATION
         ENV.setJarsPath( TOSTRING(MULTIMC_JARS_LOCATION) );
 #endif
