diff --git a/tools/actions/container_manager.py b/tools/actions/container_manager.py
index 24e410b..10b4172 100644
--- a/tools/actions/container_manager.py
+++ b/tools/actions/container_manager.py
@@ -116,11 +116,6 @@ def start(args):
         # Generate props
         make_prop(args.work + "/waydroid.prop")
 
-        # Networking
-        command = [tools.config.tools_src +
-                   "/data/scripts/waydroid-net.sh", "start"]
-        tools.helpers.run.user(args, command, check=False)
-
         # Sensors
         if which("waydroid-sensord"):
             tools.helpers.run.user(
@@ -190,11 +185,6 @@ def stop(args):
             session_cfg["session"]["state"] = helpers.lxc.status(args)
             tools.config.save_session(session_cfg)
 
-        # Networking
-        command = [tools.config.tools_src +
-                   "/data/scripts/waydroid-net.sh", "stop"]
-        tools.helpers.run.user(args, command, check=False)
-
         #TODO: remove NFC hacks
         if which("start"):
             command = ["start", "nfcd"]
